src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/MergeDataToSearchResultOperation.java

```diff
@@ -24,32 +27,49 @@ public MergeDataToSearchResultOperation(final UIDefinitionRepository uiDefinitio
         this.uiDefinitionRepository = uiDefinitionRepository;
     }
 
-    public SearchResultView execute(final CaseType caseType, final List<CaseDetails> caseDetails,final String view) {
+    public SearchResultView execute(final CaseType caseType, final List<CaseDetails> caseDetails, final String view, final String resultError) {
         final SearchResult searchResult = getSearchResult(caseType, view);
         final List<SearchResultViewColumn> viewColumns = Arrays.stream(searchResult.getFields())
             .flatMap(searchResultField -> caseType.getCaseFields().stream()
-            .filter(caseField -> caseField.getId().equals(searchResultField.getCaseFieldId()))
-            .map(caseField ->  new SearchResultViewColumn(
-                searchResultField.getCaseFieldId(),
-                caseField.getFieldType(),
-                searchResultField.getLabel(),
-                searchResultField.getDisplayOrder(),
-                searchResultField.isMetadata())
-             ))
-            .collect(Collectors.toList());
+                .filter(caseField -> caseField.getId().equals(searchResultField.getCaseFieldId()))
+                .map(caseField -> new SearchResultViewColumn(
+                    searchResultField.getCaseFieldId(),
+                    caseField.getFieldType(),
+                    searchResultField.getLabel(),
+                    searchResultField.getDisplayOrder(),
+                    searchResultField.isMetadata()))
+                )
+             .collect(Collectors.toList());
 
         final List<SearchResultViewItem> viewItems = caseDetails.stream()
-            .map(caseData -> new SearchResultViewItem(caseData.getReference().toString(), caseData.getCaseDataAndMetadata()))
+            .map(caseData -> buildSearchResultViewItem(caseData))
             .collect(Collectors.toList());
-        return new SearchResultView(viewColumns, viewItems);
+        return new SearchResultView(viewColumns, viewItems, resultError);
+    }
+
+    private SearchResultViewItem buildSearchResultViewItem(CaseDetails caseData) {
+        return new SearchResultViewItem(hasCaseReference(caseData) ? getCaseReference(caseData) : getDraftReference(caseData),
```

Comment:
to make the code more self explanatory, instead of checking hasCaseReference, we could introduce a method on CaseDetails `isDraft` which does the check. 
And here: 
```caseData.isDraft() ? caseData.getCaseReference().get() : caseData.getDraftReference()

Reply:
See above.